pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -288,28 +288,34 @@ func (a *AddVolumeOptions) Validate(isAddOp bool) error {
 				if len(a.Path) == 0 {
 					return errors.New("must provide --path for --type=hostPath")
 				}
-			case "secret":
 				if len(a.DefaultMode) > 0 {
-					if ok, _ := regexp.MatchString(`\b0?[0-7]{3}\b`, a.DefaultMode); !ok {
-						return fmt.Errorf("default-mode must be between 0000 and 0777")
-					}
+					return errors.New("--default-mode is only available for types secret and configmap")
```

Comment:
nit, maybe consider changing to something like `--default-mode is only available for secrets and configmaps`